### ISSUE TITLE
[indent] Long generic list shouldn't give a indent error.

### ISSUE THREAD

**Issue by rubenferreira97** [typescript-eslint/typescript-eslint#4438]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

.eslintrc.cjs:
```CJS
module.exports = {
  env: {
    node: true,
  },
  extends: [
    'airbnb-base',
    'airbnb-typescript/base',
  ],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 12,
    sourceType: 'module',
    project: './src/tsconfig.json',
    extraFileExtensions: ['.cjs'],
  },
  plugins: [
    '@typescript-eslint',
  ],
  rules: {
    'linebreak-style': ['error', (process.platform === 'win32' ? 'windows' : 'unix')],
    'no-console': 'off',
    '@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
  },
  ignorePatterns: ['build/', '.eslintrc.cjs'],
};
```
Typescript code example:
```TS
type T = FastifyInstance<
  Server,
  IncomingMessage,
  ServerResponse,
  FastifyLoggerInstance,
  FastifyTypeProviderDefault
>;
```

**Expected Result**

Don't show any indentation error with the following code:

``` TS
type a = FastifyInstance<
  Server,
  IncomingMessage,
  ServerResponse,
  FastifyLoggerInstance,
  FastifyTypeProviderDefault
>;
```

**Actual Result**

> Expected indentation of 0 spaces but found 2. eslint(@typescript-eslint/indent)

After auto fixing:

``` TS
type a = FastifyInstance<
Server,
IncomingMessage,
ServerResponse,
FastifyLoggerInstance,
FastifyTypeProviderDefault
>;
```

**Additional Info**
Eslint should allow manual indentation on long generic list since Eslint throws (max-len) when the LoC is too lengthy.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.9.1` |
| `@typescript-eslint/parser`        | `5.9.1` |
| `TypeScript`                       | `4.5.4` |
| `ESLint`                           | `8.6.0` |
| `node`                             | `16.13.2` |

**Comment by bradzacher** [comment]
#1824 - please use the search next time.
